### PR TITLE
chore: return Privatekey, Transaction and PublicKey legacy functionality

### LINKED ISSUE #2930: # Breaking Change Without Deprecation: `signTransaction` Return Type Changed
### Description

# Breaking Change Without Deprecation: `signTransaction` Return Type Changed

## Summary
The recent update to `@hashgraph/sdk` changed the return type of the `signTransaction` method from `Uint8Array` to `SignatureMap` without following proper deprecation protocols. This breaking change has significant negative impacts on applications using this method and requires substantial refactoring efforts.

## Issue Description
In a recent version of `@hashgraph/sdk`, the following method signature was changed:

**Before:**
```typescript
signTransaction(transaction: Transaction): Uint8Array;
```

**After:**
```typescript
signTransaction(transaction: Transaction): SignatureMap;
```

This change was implemented on the same method name without any deprecation notice, migration path, or backward compatibility support. Modern software development practices recommend maintaining backward compatibility through proper deprecation cycles, especially for public-facing APIs.

## Impact
This change has caused significant problems for our development team:

1. Unexpected runtime errors occurred immediately after upgrading the SDK
2. Our codebase required extensive refactoring to accommodate the new return type
3. Development time was diverted from feature work to fixing integration issues
4. Testing efforts had to be redirected to ensure the refactored code works correctly
5. This compromised our release schedule and increased development costs

## Suggested Solution
To maintain backward compatibility while introducing new functionality:

1. **Follow Proper Deprecation Protocol:**
   - Mark the current method as deprecated with `@deprecated` annotations
   - Document the deprecation in release notes and README
   - Keep the original method functioning alongside the new one for at least one major version cycle

2. **Add New Method Instead of Changing Existing Method:**
   ```typescript
   // Keep original method
   signTransaction(transaction: Transaction): Uint8Array;
   
   // Add new method with clear naming
   signTransactionWithSignatureMap(transaction: Transaction): SignatureMap;
   ```

3. **Define a Deprecation Timeline:**
   - Clearly communicate when the deprecated method will be removed
   - Provide migration guides for moving to the new method
   - Consider implementing console warnings when deprecated methods are used

## Best Practices for SDK Evolution
For future reference, we recommend following these practices:

1. **Semantic Versioning:** Follow SemVer strictly for all releases
2. **Change Log:** Clearly document breaking changes in CHANGELOG.md
3. **Migration Guides:** Provide step-by-step guidance for upgrading
4. **Deprecation Notices:** Use proper annotations and runtime warnings
5. **Breaking Changes:** Reserve breaking changes for major version updates

## Conclusion
Breaking changes without proper deprecation cycles create significant overhead for developers using the SDK. By implementing a proper deprecation strategy, SDK maintainers can provide new functionality while respecting the investment users have made in existing implementations.

We respectfully request that the team considers reverting this change or providing backward compatibility until proper deprecation can be implemented in a future major version release.

### Steps to reproduce

1. install v0.44.0 or older
2. create an engine/script to sign any transaction using the signTransaction method.
3. update to version 0.51.0 or higher
4. verify the back compatibility is broken, and you'll be forced to adapt your code.

### Additional context

_No response_

### Hedera network

_No response_

### Version

v0.59.0

### Operating system

None

### PR DESCRIPTION
**Pull request by ivaylonikolov7**
**Description**:
 TBD

**Related issue(s)**:
#2930

Fixes #2930

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by ivaylonikolov7** [comment]
- remove hedera references in old files

File: src/transaction/Transaction.js [hunk 1]
```diff
@@ -798,10 +799,14 @@ export default class Transaction extends Executable {
      * Add a signature explicitly
      *
      * @param {PublicKey} publicKey
-     * @param {SignatureMap} signatureMap
+     * @param {SignatureMap | Uint8Array |Uint8Array[]} signatureMap
```
[the remark this answers is not in the document]
Comment (ivaylonikolov7) [reply src/transaction/Transaction.js:802]: fair point, i will add this

File: examples/legacy-way-sign-transaction.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    Status,
+} from "@hashgraph/sdk";
+
+import dotenv from "dotenv";
+
+dotenv.config();
+
+const OPERATOR_ID = AccountId.fromString(process.env.OPERATOR_ID);
+const OPERATOR_KEY = PrivateKey.fromStringED25519(process.env.OPERATOR_KEY);
+const HEDERA_NETWORK = process.env.HEDERA_NETWORK || "testnet";
```
Comment (ivaylogarnev-limechain) [inline examples/legacy-way-sign-transaction.js:15]: Do you need the logical OR with `testnet` as the default, or could you omit it or switch it to `local` since we run most examples typically on a local network?

File: examples/mix-of-signing-transactions.js
Comment (ivaylogarnev-limechain) [inline examples/mix-of-signing-transactions.js:1]: You can include a description explaining what "mix of signing transactions" means—specifically, that we are utilizing both modern and legacy signature methods.

File: examples/mix-of-signing-transactions.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            .execute(client)
+    ).getReceipt(client);
+
+    const transferTx = new TransferTransaction()
+        .addHbarTransfer(accountId, new Hbar(-1))
+        .addHbarTransfer(recipientId2, new Hbar(1))
+        .freezeWith(client);
+
+    // Sign transaction and add the signatures to the transaction
+    const signature1 = key1.signTransaction(transferTx, true);
```
Comment (venilinvasilev) [inline examples/mix-of-signing-transactions.js:65]: I am not sure if it is a good idea to have this as an example (mixed ways of signing a transaction), but rather having it as a test to make sure this flow cannot be broken with using both approaches (as edge case that is if some one unintentionally does it). Can be a bit misleading to the user as one of the method is deprecated, meaning that it is not recommended to be used.

File: manual/MIGRATING_SIGN_TRANSACTION.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+// SignatureMap handles the signature organization
+const signatureMap = privateKey.signTransaction(transaction);
+transaction.addSignature(privateKey.publicKey, signatureMap);
+```
+
+### Why and when?
+
+## When did we make this change?
+
+-   Current Status: The change is being implemented in response to Issue (#2595)[https://github.com/hiero-ledger/hiero-sdk-js/issues/2595]
```
Comment (ivaylogarnev-limechain) [inline manual/MIGRATING_SIGN_TRANSACTION.md:93]: Fix the link syntax

File: manual/MIGRATING_SIGN_TRANSACTION.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+#### New Mode:
+
+-   **Supports signing of chunked transactions e.g FileAppend chunk transaction**
+-   Signatures are organized in a SignatureMap class
+-   Signatures are mapped to specific node IDs and transaction IDs
+-   More structured and type-safe approach
+
+## Important Considerations
+
+1. Backward Compatibility - Legacy mode still works
```
Comment (ivaylogarnev-limechain) [inline manual/MIGRATING_SIGN_TRANSACTION.md:31]: I think the wording can be improved here.

File: examples/mix-of-signing-transactions.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    const signature2 = key2.signTransaction(transferTx);
```
Comment (0xivanov) [inline examples/mix-of-signing-transactions.js:66]: It's hard to understand which way is legacy and which not. Can add more code comments and change the variable names to make it more clear.

File: manual/MIGRATING_SIGN_TRANSACTION.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
+# Migration Guide: Transaction Signing in Hedera SDK
+
+## Overview
+
+The Hedera SDK has updated its transaction signing mechanism to provide better type safety and a more structured approach to managing signatures. This guide will help you migrate from the legacy mode to the new mode.
```
Comment (0xivanov) [inline manual/MIGRATING_SIGN_TRANSACTION.md:5]: Can add that the new API is to work with multinode chunked transactions

**Review by venilinvasilev — APPROVED** [review]
LGTM

**Review by ivaylogarnev-limechain — DISMISSED** [review]
LGTM

**Review by 0xivanov — APPROVED** [review]
LGTM

**Review by venilinvasilev — APPROVED** [review]
